sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetReadState.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.parquet;
+
+/**
+ * Helper class to store intermediate state while reading a Parquet column chunk.
+ */
+final class ParquetReadState {
+  /** Maximum definition level */
+  int maxDefinitionLevel;
+
+  /** The offset to add the next value in the current batch */
+  int offset;
```

Comment:
Shall we have an initial value, `-1` and use it in`advanceOffset` to validate `resetForBatch` is called properly before?

Reply:
That's somewhat useful but it adds a branching in a hot path. Also we may have to remember reset the offset to -1 after reading a batch. Because these are internal methods, I guess it's not easy to go wrong without being captured by unit tests and other things.